pkg/automationconfig/automation_config_builder.go

```diff
@@ -156,6 +162,9 @@ func (b *Builder) AddVersion(version MongoDbVersionConfig) *Builder {
 
 func (b *Builder) SetMongoDBVersion(version string) *Builder {
 	b.mongodbVersion = version
+	if b.isEnterprise {
```

Comment:
I'm scratching my head if this is the best approach. Modifying something in a setter is mostly considered as a bad practice, but this is really what is happening here - the proper version for enterprise has the `-ent` suffix. Without it, it's considered as community, which does not reflect the true. 

That was the main argument for doing this in a setter. WDYT?

Reply:
As long as there is no different way to mark a image as enterprise (i.e. through a setting) I think thats ok

Reply:
This still bothers me - let me rework this piece of the code.

Reply:
Addressed. Closing.